### PR TITLE
Add memory tracking to the new Parquet reader

### PR DESCRIPTION
**Pull request by nezihyigitbasi**

### CODE REVIEW

**Comment by nezihyigitbasi** [comment]
@zhenxiao can you also take a look?

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java [hunk 1]
```diff
@@ -125,6 +138,14 @@ public Block readBlock(ColumnDescriptor columnDescriptor, Type type)
         return columnReader.readBlock(type);
     }
 
+    private byte[] allocateBlock(int length)
+    {
+        byte[] buffer = new byte[length];
+        LocalMemoryContext blockMemoryContext = currentRowGroupMemoryContext.newLocalMemoryContext();
+        blockMemoryContext.setBytes(buffer.length);
```
Comment (dain) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java]: This doesn't look correct.  The `setBytes` method should be setting the currently used memory of the whole context, so each time you allocate a new block you it clears the previous allocation.  Also, the buffer appears to be a temporary variable and not "held" by the reader, so it shouldn't be declared as system memory.

Comment (nezihyigitbasi) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java]: I was thinking like, each time `advanceToNextRowGroup()` is called a new row group gets allocated and that's accounted for in the `blockMemoryContext` (each row group causes a large `buffer` allocation) and it gets cleared with the next `advanceToNextRowGroup()` call. So the memory usage of the reader will look like a pulse going up and down as row groups get allocated/deallocated.  The `buffer` is effectively held by the reader until the next `advanceToNextRowGroup()` call.

What's the right way to capture this?

Comment (nezihyigitbasi) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetReader.java]: @dain any comments on this one? `OrcInputStream` is doing something similar with its `bufferMemoryUsage`.

**Comment by dain** [comment]
Merged, thanks!

**Pull request closed without merging**